src/SignalR/clients/ts/FunctionalTests/ts/HubConnectionTests.ts

```diff
@@ -679,14 +679,19 @@ describe("hubConnection", () => {
             if (transportType !== HttpTransportType.LongPolling) {
                 it("terminates if no messages received within timeout interval", async (done) => {
                     const hubConnection = getConnectionBuilder(transportType).build();
-                    hubConnection.serverTimeoutInMilliseconds = 100;
 
                     hubConnection.onclose((error) => {
                         expect(error).toEqual(new Error("Server timeout elapsed without receiving a message from the server."));
                         done();
                     });
 
                     await hubConnection.start();
+
+                    // set this after start completes to avoid network issues with the handshake taking over 100ms and causing a failure
+                    hubConnection.serverTimeoutInMilliseconds = 1;
+
+                    // invoke a method with a response to reset the timeout using the new value
+                    await hubConnection.invoke("Echo", "");
```

Comment:
Too bad we have to call another method to observe the change to serverTimeoutInMilliseconds.